server/lib/chat/chat-service.test.ts

```diff
@@ -396,6 +396,79 @@ describe('chat/chat-service', () => {
     clearTestLogs(nydus)
   })
 
+  describe('handleNewUser', () => {
+    test('subscribes user to their joined channels', async () => {
+      const user3: SbUser = { id: 3 as SbUserId, name: 'USER_NAME_3' }
+      const user3ShieldBatteryChannelEntry = { ...user1ShieldBatteryChannelEntry, userId: user3.id }
+      const user3TestChannelEntry = { ...user1TestChannelEntry, userId: user3.id }
```

Comment:
You should adjust the times on these channel entries (in the constant you're pulling from here) imo, this isn't really realistic atm (they have the same join time, and that's what we'd be sorting on). It's possible in the future we'd need to sort this data in the server code or something, in which case this test could break even if the server code was actually correct.